apps/jobs/views.py

```diff
@@ -624,6 +625,152 @@ def leaderboard(request, challenge_phase_split_id):
     return paginator.get_paginated_response(response_data)
 
 
+@api_view(["GET"])
+@throttle_classes([AnonRateThrottle])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_all_entries_on_public_leaderboard(request, challenge_phase_split_pk):
+    """
+    Returns public/private leaderboard entries to corresponding challenge phase split for a challenge host
+
+    - Arguments:
+        ``challenge_phase_split_pk``: Primary key for the challenge phase split for which leaderboard is to be fetched
+
+    - Returns:
+        All Leaderboard entry objects in a list
+    """
+    # check if the challenge exists or not
+    challenge_phase_split = get_challenge_phase_split_model(challenge_phase_split_pk)
+
```

Comment:
Please add the check for access only right after this line, so as to reduce the database call. If the user is not a host then we should do minimum possible queries and return the error.
```
    challenge_obj = challenge_phase_split.challenge_phase.challenge

    # Allow access only to challenge host
    if not is_user_a_host_of_challenge(request.user, challenge_obj.pk):
        response_data = {"error": "Sorry, you are not authorized to make this request!"}
        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
```